internal/tasks/config_helpers.go

```diff
@@ -4,46 +4,52 @@ import (
 	"fmt"
 	"regexp"
 	"strings"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
 )
 
 var (
 	paramsRegex *regexp.Regexp = regexp.MustCompile(`(?P<full>{{\w*(?P<param>.*?)\w*}})`)
 	labelRegex  *regexp.Regexp = regexp.MustCompile(`device.metadata.labels\[(?P<key>.*)\]`)
+	nameRegex   *regexp.Regexp = regexp.MustCompile(`device.metadata.name`)
 )
 
 func ContainsParameter(b []byte) bool {
 	return paramsRegex.Match(b)
 }
 
-func ReplaceParameters(b []byte, labels *map[string]string) ([]byte, error) {
+func ReplaceParameters(b []byte, objectMeta api.ObjectMeta) ([]byte, error) {
 	replacements := map[string]string{}
 	paramsToMatches := map[string]string{}
 
-	if labels == nil {
-		return b, nil
-	}
-
 	matches := paramsRegex.FindAllStringSubmatch(string(b), -1)
 	for _, match := range matches {
 		paramsToMatches[match[0]] = match[1]
 	}
 
 	for param, match := range paramsToMatches {
-		if !labelRegex.MatchString(param) {
+		switch {
+		case nameRegex.MatchString(param):
+			if objectMeta.Name == nil {
+				return []byte(""), fmt.Errorf("parameter referenced name, but no name found")
```

Comment:
optional: it's common to return nil if a function fails to generate a result
```suggestion
				return nil, fmt.Errorf("parameter referenced name, but no name found")
```